gdsfactory/pdk.py

```diff
@@ -560,7 +552,7 @@ def get_cross_section_name(self, cross_section: CrossSection) -> str:
             (
                 key
                 for key, value in self.cross_sections.items()
-                if value == cross_section
+                if value() == cross_section
```

Comment:
Just want to note that this is the reason the tests are failing, just wanted to raise this to see if maybe this was wrong before. Since the values in self.cross_sections are all functions (and assuming this is actually true in practice), this should never find a equal cross section in the pdk, so it will always return cross_section.name.

Reply:
yes, maybe it was wrong before, nice catch